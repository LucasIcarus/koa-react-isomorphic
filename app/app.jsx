/* @flow */
/* global process */
import "./client/helpers/loadExternalLibs";
import getRouter from "./client/helpers/getRouter";
import initialize from "./client/helpers/initialize";

(async () => {
  const appDOM = document.getElementById("app");

  if (!appDOM) {
    throw new Error("Cannot initialise application");
  }

  initialize(await getRouter(), appDOM);

  if (process.env.NODE_ENV === "development" && module.hot) {
    // $FlowFixMe
    module.hot.accept("./client/helpers/getRouter", async () => {
      initialize(await require("./client/helpers/getRouter").default(), appDOM);
    });
  }
})();

if (process.env.NODE_ENV === "production") {
  const runtime = require("offline-plugin/runtime");

  runtime.install({
    onUpdating: () => {
      console.log("SW Event:", "onUpdating"); // eslint-disable-line
    },
    onUpdateReady: () => {
      console.log("SW Event:", "onUpdateReady"); // eslint-disable-line
      // Tells to new SW to take control immediately
      runtime.applyUpdate();
    },
    onUpdated: () => {
      console.log("SW Event:", "onUpdated"); // eslint-disable-line
      // Reload the webpage to load into the new version
      window.location.reload();
    },

    onUpdateFailed: () => {
      console.log("SW Event:", "onUpdateFailed"); // eslint-disable-line
    }
  });
}
