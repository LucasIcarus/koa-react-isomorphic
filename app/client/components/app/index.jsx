/* @flow */
/* global process */
import React from "react";
import Helmet from "../helmet";

export default ({ router }: { router: Object}) => {
  let Component = (
    <div>
      <Helmet />
      {router}
    </div>
  );

  if (process.env.NODE_ENV === "development") {
    const { AppContainer } = require("react-hot-loader");

    Component = (
      <AppContainer>
        {Component}
      </AppContainer>
    );
  }

  return Component;
};
