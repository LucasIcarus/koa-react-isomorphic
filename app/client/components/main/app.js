import React from 'react';
import { Provider } from 'react-redux';
import Debug from './debug';

export default ({ store, routes }) => (
  <Provider key='provider' store={ store }>
    <div>
      { routes }
      <Debug />
    </div>
  </Provider>
);
