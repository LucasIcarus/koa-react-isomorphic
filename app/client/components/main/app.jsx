import React from 'react';
import { Provider } from 'react-redux';
import Helmet from 'client/components/helmet';

const App = ({ store, routes }: { store: Object, routes: Object }) => {
  if (process.env.NODE_ENV === 'development') {
    const { AppContainer } = require('react-hot-loader');

    return (
      <AppContainer>
        <Provider key="provider" store={store}>
          <div>
            <Helmet />
            {routes}
          </div>
        </Provider>
      </AppContainer>
    );
  }

  return (
    <Provider key="provider" store={store}>
      <div>
        <Helmet />
        {routes}
      </div>
    </Provider>
  );
};

export default App;
