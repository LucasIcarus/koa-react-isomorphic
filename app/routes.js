import React from 'react';
import { Router, Route } from 'react-router';
import { ReduxAsyncConnect } from 'redux-async-connect';
import Todos from './client/components/templates/todos';

const history = process.env.RUNTIME_ENV === 'client'
                  ? require('react-router').browserHistory
                  : require('history/lib/createLocation')();

function render(props) {
  return <ReduxAsyncConnect { ...props } />;
}

export default (
  <Router history={ history } render={ render }>
    <Route path='/' component={ Todos } />
  </Router>
);
