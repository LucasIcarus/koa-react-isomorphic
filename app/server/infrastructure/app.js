/* @flow */
import Koa from "koa";
import {
  apiLayer,
  assetsLayer,
  errorLayer,
  initialLayer,
  loggingLayer,
  renderLayer,
  securityLayer
} from "./middlewares";
import apis from "../application/apis";
import controllers from "../application/controllers";

const app = new Koa();

// setup middlewares
loggingLayer(app);
errorLayer(app);
assetsLayer(app);
securityLayer(app);
initialLayer(app);
apiLayer(app, apis);
renderLayer(app, controllers);

export default app;
