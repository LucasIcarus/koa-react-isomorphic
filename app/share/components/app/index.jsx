/* @flow */
/* global process */
import React from "react";

export default ({ router }: { router: Object }) => {
  let Component = <div>{router}</div>;

  if (process.env.NODE_ENV === "development") {
    const { AppContainer } = require("react-hot-loader");

    Component = <AppContainer>{Component}</AppContainer>;
  }

  return Component;
};
