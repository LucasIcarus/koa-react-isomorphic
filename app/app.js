import 'babel-polyfill';
import './client/lib/index';

import $ from 'jquery';
import React from 'react';
import ReactDOM from 'react-dom';
import App from 'app/client/components/main/app';
import routes from './routes';
import configureStore from './client/stores/index';

function render(store, appDOM) {
  ReactDOM.render(<App store={store} routes={routes} />, appDOM);
}

$(document).ready(() => {
  const appDOM = document.getElementById('app');
  const store = configureStore(window.__data);

  if (process.env.NODE_ENV === 'development' && !process.env.SERVER_RENDERING) {
    const { clientFetchData } = require('./client/helpers/fetch-data');

    clientFetchData(routes, store.getState().routing.location, store).then(() => {
      render(store, appDOM);
    });
  } else {
    render(store, appDOM);
  }
});
