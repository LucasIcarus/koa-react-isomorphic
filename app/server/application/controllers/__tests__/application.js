import supertest from "supertest";
import app from "../../../infrastructure/app";

describe("Controller: application", () => {
  it("should render single page", async () => {
    const result = await supertest(app.listen())
      .get("/")
      .set("Accept", "text/html")
      .set("X-CSRF-Token", process.env.SECRET_KEY.toString("base64"))
      .expect(200);

    expect(result.text).toBeDefined();
    expect(result.text).toContain(`id="app"`);
  });
});
