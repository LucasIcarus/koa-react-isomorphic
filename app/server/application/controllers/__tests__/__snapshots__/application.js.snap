// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller: application should render single page 1`] = `
"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\"><meta name=\\"theme-color\\" content=\\"#000000\\"><title>Application</title><link rel=\\"manifest\\" href=\\"/manifest.json\\"><link href=\\"/favicon.ico\\" rel=\\"shortcut icon\\" type=\\"image/x-icon\\"></head><body><div id=\\"app\\"></div><script type=\\"text/javascript\\">
    window.javascriptAssets = {};
</script><script type=\\"text/javascript\\"></script><script type=\\"text/javascript\\"></script><script type=\\"text/javascript\\"></script></body></html>"
`;
